rocmPackages.llvm: remove workaround for LLVMgold-path.patch

LunNova opened the pull request:
Prior to overrideScope support for llvmPackages, LLVMgold-path.patch would point at a non-overridden LLVM and needed replaced.

Now that we use overrideScope we can drop this jank!

This should hopefully resolve the issue where rocmPackages didn't work with CA derivations.  
See also https://github.com/NixOS/nixpkgs/pull/439003

cc @timschumi 

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

LunNova committed:
rocmPackages.llvm: remove workaround for LLVMgold-path.patch

Prior to overrideScope support for llvmPackages, LLVMgold-path.patch
would point at a non-overridden LLVM and needed replaced.

Now that we use overrideScope we can drop this patch.

emilazy commented on pkgs/development/rocm-modules/6/llvm/default.nix:
```diff
@@ -213,12 +213,7 @@ let
-      (lib.strings.hasSuffix "add-nostdlibinc-flag.patch" (baseNameOf x))
-      || (lib.strings.hasSuffix "clang-at-least-16-LLVMgold-path.patch" (baseNameOf x))
-    );
+  removeInapplicablePatches = x: (lib.strings.hasSuffix "add-nostdlibinc-flag.patch" (baseNameOf x));
```
This patch no longer exists; it was moved to the wrapper.

LunNova replied:
Nice!

LunNova replied:
Dropped `removeInapplicablePatches`

LunNova committed:
rocmPackages.llvm: remove unnecessary filtering of nostdlibinc patch

add-nostdlibinc-flag.patch is no longer needed as it was moved to
the cc-wrapper
